### PR TITLE
Use django-rest-framework to prepare for using React as frontend

### PR DESCRIPTION
**Pull request by hunj**
# _𝕞𝕠𝕕𝕖𝕣𝕟 𝕗𝕣𝕒𝕞𝕖𝕨𝕠𝕣𝕜_

## Overview

This PR: 

- Adds a `web` docker container that runs nodejs and React.
  - Currently it's just the "hello world" page of React. Will need @hafley66's input for additional config.
- Adds a `nginx` docker container that will route incoming (port 80) request to `web` container
  (NOTE: the current config is only prod-like. We'll need to figure out the "devmode" to this, in which `web` would run something like `yarn dev`?)
- Introduces `django-rest-framework` and related packages, i.e. `rest_auth`.
- Currently has API endpoints (you can view them on a web-based interface on devmode, i.e. `DEBUG=True`)
  - http://localhost:8000/api/ - just general overview, doesn't really do much
  - http://localhost:8000/api/hackathons/ - GETs all `Hackathon` objects
  - http://localhost:8000/api/auth/login/ - This is where you POST login creds, it will return `key` JSON object upon succ
  - Refer to [django-rest-auth documentation](https://django-rest-auth.readthedocs.io/en/latest/api_endpoints.html) for other auth-related endpoints

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
       DJANGO_SECRET_KEY: /var/run/secrets/django_secret
       EMAIL_SERVICE_NAME: mailchimp
       EMAIL_NEWLETTER_LIST_ID: d574cb485f
+      DATABASE_HOST: db
     secrets:
       - django_secret
 
+  web:
+    container_name: web
+    image: node:15.8.0
```
Comment (hafley66) [inline docker-compose.yml:22]: i would do node 14, odd numbers are unstable and very bleeding edge

File: docker-compose.yml [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
+      - ./web:/web
+
+  nginx:
+    container_name: nginx
+    image: nginx:latest
+    ports:
+      - 80:8080
+    volumes:
+      - ./nginx/nginx-proxy.conf:/etc/nginx/conf.d/default.conf:ro
+      - ./web/build:/var/www/frontend # maps frontend build inside nginx
```
Comment (hafley66) [inline docker-compose.yml:35]: hmm, in dev mode we should be using webpack dev server? that would mean we just need nginx to route to that server's port. running webpack in docker is known problem, large amount of file IO is not docker+non-linux strongpoint. i usually end up using cross-env to support running simple shell commands on windows and mac to run webpack in host. 

dont get me wrong, docker is important for repeatable builds, but in prod, the assets would be compiled using multi-stage docker build, and uploaded into nginx/cdn folder

File: docker-compose.yml (hunk 1 above)
Comment (hunj) [reply docker-compose.yml:22]: Gotcha, this is exactly the kind of help I needed -- I don't know much about nodejs ecosystem. 
Any specific minor version preference?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
-  web:
-    container_name: hackathonmentors
-    build: .
+  django:
+    container_name: api
+    build: ./api
     command: python hackathonmentors/manage.py runserver 0.0.0.0:8000
     volumes:
-      - .:/build
+      - ./api:/build
```
[the remark this answers is not in the document]
Comment (hunj) [reply docker-compose.yml:9]: Was probably something from tutorial and whatnot. `/app` or `/api` makes more sense as you said

File: docker-compose.yml (hunk 2 above)
Comment (hunj) [reply docker-compose.yml:35]: Yeah I wasn't able to really figure out setting up hot-reloading and whatnot for dev mode. Could use some help on this part 🐙

**Pull request merged**